docs/building-a-form/common-patterns-for-building-forms.md

```diff
@@ -151,64 +151,52 @@ To create a block of text with no fields that follow, create an empty view objec
 
 ### Conditionally hiding a group of fields
 
-Sometimes fields in a form are siblings to others, but must be hidden conditionally. For example, see this schema snippet from [VA Form 22-5490](https://www.va.gov/vaforms/form_detail.asp?FormNo=22-5490):
+Sometimes fields in a form are siblings to others, but should be hidden conditionally. For example, this schema defines a field named `employed` that is rendered as a checkbox the user can check. If this field is false (the checkbox is not checked) there is no need to display the `jobStartDate` or `monthlyWages` fields:
 
 ```json
-"previousBenefits": {
-  "type": "object",
-  "properties": {
-    "disability": { "type": "boolean" },
-    "dic": { "type": "boolean" },
-    "chapter31": { "type": "boolean" },
-    "ownServiceBenefits": { "type": "string" },
-    "chapter35": { "type": "boolean" },
-    "chapter33": { "type": "boolean" },
-    "transferOfEntitlement": { "type": "boolean" },
-    "other": { "type": "string" },
-    "veteranFullName": { "$ref": "#/definitions/fullName" },
-    "veteranSocialSecurityNumber": { "$ref": "#/definitions/ssn" }
+schema: {
+  type: 'object',
+  properties: {
+    employed: { type: 'boolean' },
+    jobStartDate: { type: 'string' },
+    monthlyWages: { type: 'string' },
+    otherMonthlyIncome: { type: 'string' }
   }
 }
+
 ```
 
-Only `chapter35`, `chapter33`, `transferOfEntitlement`, `veteranFullName`, and `veteranSocialSecurityNumber` are conditionally hidden, so the `schema` and `uiSchema` are written as:
+To accomplish this, the `schema` and `uiSchema` are written as:
 
 ```js
-// schema
-{
-  disability: { ... },
-  dic: { ... },
-  chapter31: { ... },
-  ownServiceBenefits: { ... },
-  'view:sponsorServiceOptions': {
-    chapter35: { ... },
-    chapter33: { ... },
-    transferOfEntitlement: { ... },
-    veteranFullName: { ... },
-    veteranSocialSecurityNumber: { ... }
-  },
-  other: { ... }
-}
-
-// uiSchema
-{
-  disability: { ... },
-  dic: { ... },
-  chapter31: { ... },
-  ownServiceBenefits: { ... },
-  'view:sponsorServiceOptions': {
-    hideIf: (formData) => /* Some condition here */,
-    chapter35: { ... },
-    chapter33: { ... },
-    transferOfEntitlement: { ... },
-    veteranFullName: { ... },
-    veteranSocialSecurityNumber: { ... }
+schema: {
+  type: 'object',
+  properties: {
+    employed: { type: 'boolean' },
+    'view:jobInformation': {
+      type: 'object',
+      properties: {
+        jobStartDate: { type: 'string' },
+        monthlyWages: { type: 'string' }
+      }
+    },
+    otherMonthlyIncome: { type: 'string' }
+  }
+},
+uiSchema: {
+  employed: { 'ui:title': 'I am employed' },
+  'view:jobInformation': {
+    'ui:options': {
+      hideIf: (formData) => !formData.employed
```

Comment:
Something I'm not totally clear on is the difference between `hideIf` and `expandUnder`. I think from reading [this function](https://github.com/usds/us-forms-system/blob/master/src/js/state/helpers.js#L98) that they are used to set 2 different `uiSchema` properties: `hideIf` sets `'ui:hidden'` and `expandUnder` sets `'ui:collapsed'`. Functionally, though, they seem to do the same thing. I think there are some benefits to using `expandUnder` if you want [to style](https://github.com/usds/us-forms-system/blob/master/docs/building-a-form/common-patterns-for-building-forms.md#styling-expanded-or-collapsed-fields) the expanded fields in a certain way, but otherwise I don't know if there's a big difference?

Reply:
Honestly I don't know, the [logic looks nearly identical](https://github.com/usds/us-forms-system/blob/4d69ee604610de969e3d8d80f62fafabf87aa428/src/js/state/helpers.js#L113-L129) and [we consider them both hidden](https://github.com/usds/us-forms-system/blob/4d69ee604610de969e3d8d80f62fafabf87aa428/src/js/state/helpers.js#L9-L11). Other than that code I don't see anything that uses or examines it, even in RJSF?